src/celeritas/decay/interactor/MuDecayInteractor.hh

```diff
@@ -0,0 +1,299 @@
+template<class Engine>
+CELER_FUNCTION Interaction MuDecayInteractor::operator()(Engine& rng)
+{
+    // Allocate secondaries
+    Secondary* secondaries = allocate_(3);
+    if (secondaries == nullptr)
+    {
+        // Failed to allocate secondaries
+        return Interaction::from_failure();
+    }
+
+    real_type electron_energy_frac;
+    real_type electron_nu_energy_frac;
+    size_type const max_iter{1000};  // \todo Why?
+
+    for ([[maybe_unused]] auto i : range(max_iter))
+    {
+        for ([[maybe_unused]] auto j : range(max_iter))
+        {
+            electron_nu_energy_frac = sample_max_ * generate_canonical(rng);
+            if (generate_canonical(rng)
+                <= electron_nu_energy_frac
+                       * (real_type{1} - electron_nu_energy_frac))
+            {
+                break;
+            }
+            electron_nu_energy_frac = sample_max_;
+        }
+
+        electron_energy_frac = generate_canonical(rng);
+        if (electron_nu_energy_frac >= real_type{1} - electron_energy_frac)
+        {
+            break;
+        }
+        electron_nu_energy_frac = real_type{1} - electron_energy_frac;
+    }
+    real_type muon_nu_energy_frac = real_type{2} - electron_energy_frac
+                                    - electron_nu_energy_frac;
+
+    // Momentum of secondaries at rest frame; neutrino masses are ignored
+    auto charged_lep_energy
+        = this->momentum(electron_energy_frac, shared_.electron_mass);
+    auto electron_nu_energy = this->momentum(electron_nu_energy_frac, 0);
+    auto muon_nu_energy = this->momentum(muon_nu_energy_frac, 0);
+
+    // Angle between charged lepton and electron neutrino
+    real_type cos_theta
+        = real_type{1} - 2 / electron_energy_frac - 2 / electron_nu_energy_frac
+          + 2 / (electron_nu_energy_frac * electron_energy_frac);
+    CELER_ASSERT(std::fabs(cos_theta) <= 1);
+    real_type sin_theta = std::sqrt(real_type{1} - ipow<2>(cos_theta));
+
+    // Use an initial arbitrary direction for secondaries.
+    // Placing the electron on one axis (e.g. +z), simplifies the calculation
+    // for the other two secondaries. The direction of the electron neutrino
+    // is defined by the sampled theta, and the muon neutrino direction is
+    // calculated via conservation of momentum
+    Real3 charged_lep_dir = {0, 0, 1};
+    Real3 electron_nu_dir = {sin_theta, 0, cos_theta};
+    Real3 muon_nu_dir
+        = {-(electron_nu_energy_frac / muon_nu_energy_frac) * sin_theta,
+           0,
+           -electron_energy_frac / muon_nu_energy_frac
+               - (electron_nu_energy_frac / muon_nu_energy_frac) * cos_theta};
+
+    // Apply a spherically uniform rotation to the decay
+    real_type euler_phi
+        = UniformRealDistribution<real_type>(0, 2 * constants::pi)(rng);
+    real_type euler_theta = std::acos(2 * generate_canonical(rng) - 1);
+    real_type euler_psi
+        = UniformRealDistribution<real_type>(0, 2 * constants::pi)(rng);
```

Comment:
Probably, define
`using UniformRealDist = UniformRealDistribution<real_type>; `
in the class type alias and use UniformRealDist in other all places too.